### PR TITLE
fix(menu): add className to menu component

### PR DESCRIPTION
**Pull request by Lawndlwd**
## Type

- Bug

#### What is expected?

Ability to add styled to  `<Menu>` component  ex: ```styled(Menu) more style here```

#### The following changes where made:

(Describe what you did)

1. add className to menu props

2. pass className to Popper inside the Menu

3.apply styled on menuList instead of css attribute

### CODE REVIEW

File: src/components/Menu/index.tsx [hunk 1]
```diff
@@ -99,43 +103,39 @@ type AlignStyle = {
...
-    text-align: center;
-    justify-content: center;
-    color: ${theme.colors.neutral.text};
-    border-radius: 4px;
-    position: relative;
+const MenuList = styled.div<MenuListProps>`
+  &:after,
+  &:before {
+    left: ${({ align }) => align.left};
+    right: ${({ align }) => align.left};
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:110]:
```suggestion
    right: ${({ align }) => align.right};
```

File: src/components/Menu/index.tsx [hunk 2]
```diff
@@ -163,6 +164,7 @@ const Menu: MenuType = ({
   name = 'menu',
   placement = 'bottom',
   visible = false,
+  className = '',
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:167]:
```suggestion
  className,
```

Not needed :)

File: src/components/Menu/index.tsx [hunk 3]
```diff
@@ -173,21 +175,22 @@ const Menu: MenuType = ({
...
+          align={align}
+          hasArrow={hasArrow}
           role="menu"
-          css={[
-            styles.menu,
-            hasArrow && arrowPlacementStyles[localPlacement as ArrowPlacement],
-            styles.align(align),
-          ]}
+          css={
+            hasArrow && arrowPlacementStyles[localPlacement as ArrowPlacement]
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:187]: Should be handled in the styled component

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Menu/index.tsx [hunk 4]
```diff
@@ -1,158 +1,107 @@
...
   &:after {
     border-color: transparent;
+    border-bottom-color: ${({ theme, placement }) =>
+      placement.includes('bottom') && theme.colors.primary.borderStrong};
+    border-top-color: ${({ theme, placement }) =>
+      placement.includes('top') && theme.colors.primary.borderStrong};
+    margin-left: ${({ placement }) => placement.includes('end') && '-9px'};
+    right: ${({ placement }) => placement.includes('end') && '24px'};
+    margin-right: ${({ placement }) => placement.includes('start') && '-9px'};
+    left: ${({ placement }) => placement.includes('start') && '24px'};
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:74]: Ah euh, this seems a lot more complicated than before, we could have the same behaviour as before, but move the computing of the variant inside the styled :)

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Menu/index.tsx [hunk 5]
```diff
@@ -173,21 +175,20 @@ const Menu: MenuType = ({
...
     visible={visible}
+    className={className}
   >
     {({ placement: localPlacement, toggle, visible: isOpen }) =>
       isOpen && (
-        <div
+        <MenuList
+          align={align}
+          hasArrow={hasArrow}
+          placement={placement}
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:185]:
```suggestion
          placement={localPlacement}
```

File: src/components/Menu/index.tsx [hunk 6]
```diff
@@ -148,10 +115,44 @@ type MenuProps = Omit<ComponentProps<typeof Popper>, 'children'> & {
...
+  background-color: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+  display: flex;
+  flex-direction: column;
+  text-align: center;
+  justify-content: center;
+  color: ${({ theme }) => theme.colors.neutral.text};
+  border-radius: 4px;
+  position: relative;
+  ${({ placement, theme }) =>
+    placement && arrowPlacementStyles[placement](theme)}
```
Comment (chambo-e) [inline src/components/Menu/index.tsx:153]:
```suggestion
  ${({ placement, theme }) =>
    hasArrow && arrowPlacementStyles[placement]?.(theme)}
```

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Menu/index.tsx (hunk 6 above)
Comment (matthprost) [inline src/components/Menu/index.tsx:153]: Please use ternary

File: src/components/Menu/index.tsx [hunk 7]
```diff
@@ -173,21 +175,20 @@ const Menu: MenuType = ({
     name={name}
     placement={placement}
     visible={visible}
+    className={className}
   >
     {({ placement: localPlacement, toggle, visible: isOpen }) =>
       isOpen && (
```
Comment (matthprost) [inline src/components/Menu/index.tsx:181]: Here too if we can have a ternary

```js
isOpen ? (doStuff...) : null
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Menu/index.tsx (hunk 6 above)
Comment (chambo-e) [reply src/components/Menu/index.tsx:153]: Not required in styled :)

File: package.json [hunk 8]
```diff
@@ -98,7 +98,7 @@
     "@babel/preset-env": "7.16.11",
     "@babel/preset-react": "7.16.7",
     "@babel/preset-typescript": "7.16.7",
-    "@commitlint/cli": "16.2.1",
+    "@commitlint/cli": "16.2.3",
```
Comment (chambo-e) [inline package.json:101]: Bad rebase ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (Lawndlwd) [reply package.json:101]: it has been made automatically after rebase, so what you mean by bad rebase ? I have to revert the change on package.json

Comment (chambo-e) [reply package.json:101]: Yes, I don't think you are up to date with main because this should not show up :)

File: package.json [hunk 9]
```diff
@@ -109,7 +109,7 @@
     "@rollup/plugin-babel": "5.3.1",
     "@rollup/plugin-node-resolve": "13.1.3",
     "@rollup/plugin-url": "6.1.0",
-    "@scaleway/eslint-config-react": "3.4.8",
+    "@scaleway/eslint-config-react": "3.4.9",
```
Comment (chambo-e) [inline package.json:112]: Still an issue here :)

File: yarn.lock [hunk 10]
```diff
@@ -1713,8 +1713,8 @@ __metadata:
   linkType: hard
 
 "@commitlint/load@npm:>6.1.1, @commitlint/load@npm:^16.2.1":
-  version: 16.2.1
-  resolution: "@commitlint/load@npm:16.2.1"
+  version: 16.2.3
```
Comment (chambo-e) [inline yarn.lock:1716]: Need to revert the lock file too :)

**Review by chambo-e — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**